xlb/distribute/distribute.py

```diff
@@ -85,3 +82,38 @@ def _wrapped_operator(*args):
         return distributed_operator(*args)
 
     return jit(_wrapped_operator)
+
+
+def distribute(operator, grid, velocity_set, num_results=1, ops="permute"):
+    """
+    Distribute an operator or a stepper.
+    If the operator is a stepper, check for post-streaming boundary conditions
+    before deciding how to distribute.
+    """
+    if isinstance(operator, IncompressibleNavierStokesStepper):
+        # Check for post-streaming boundary conditions
+        has_post_streaming_bc = any(
+            bc.implementation_step == ImplementationStep.STREAMING
+            for bc in operator.boundary_conditions
+        )
+
+        if has_post_streaming_bc:
+            # If there are post-streaming BCs, only distribute the stream operator
+            distributed_stream = distribute_operator(
+                operator.stream, grid, velocity_set
+            )
+            operator.stream = distributed_stream
+        else:
+            # If no post-streaming BCs, distribute the whole operator
+            distributed_op = distribute_operator(
+                operator, grid, velocity_set, num_results=num_results, ops=ops
+            )
+            return distributed_op
+
+        return operator
+    else:
+        # For other operators, apply the original distribution logic
+        distributed_op = distribute_operator(
+            operator, grid, velocity_set, num_results=num_results, ops=ops
```

Comment:
Please move this to be the defaul results outside of the if statement and deleve else of the inner if statement.

Reply:
there is no need to spend time compiling a stepper that we won't need in the future.